src/restruct_poc/solver/constraint.hpp

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include "src/restruct_poc/model/node.hpp"
+
+namespace openturbine {
+
+enum ConstraintType {
+    Rigid,
+    FixedBC,
+    PrescribedBC,
+    Cylindrical,
+    RotationControl,
+};
```

Comment:
We should use enum class as opposed to a plain enum just to be more safe: https://stackoverflow.com/questions/18335861/why-is-enum-class-considered-safer-to-use-than-plain-enum

In addition, a little bit of description might be helpful to delineate between the types.

```
enum class ConstraintType {
    Rigid,    //< rigid body constraint
    FixedBC,
    PrescribedBC,
    Cylindrical,
    RotationControl,
};
```

Reply:
Good call on the `enum class`. I wasn't familiar with it.